tests/e2e/vsphere_volume_expansion_staging.go

```diff
@@ -0,0 +1,1331 @@
+						fmt.Sprintf("error updating  size for statefulset. PVCName: %s pvcSizeAfterExpansion: %v "+
+							"pvcSizeBeforeExpansion: %v", pvclaim.Name, pvcSizeAfterExpansion, pvcSizeBeforeExpansion))
+					ginkgo.By("File system resize finished successfully")
+
+				}
+			}
+		}
+
+		replicas = 0
+		ginkgo.By(fmt.Sprintf("Scaling down statefulsets to number of Replica: %v", replicas))
+		_, scaledownErr := fss.Scale(client, statefulset, replicas)
+		gomega.Expect(scaledownErr).NotTo(gomega.HaveOccurred())
+		fss.WaitForStatusReplicas(client, statefulset, replicas)
+		ssPodsAfterScaleDown := fss.GetPodList(client, statefulset)
+		gomega.Expect(len(ssPodsAfterScaleDown.Items) == int(replicas)).To(gomega.BeTrue(),
+			"Number of Pods in the statefulset should match with number of replicas")
+
+		framework.Logf("Deleting statefulset %v", statefulset.Name)
+		fss.WaitForStatusReadyReplicas(client, statefulset, 0)
+		err = client.AppsV1().StatefulSets(statefulset.Namespace).Delete(context.TODO(),
+			statefulset.Name, metav1.DeleteOptions{OrphanDependents: new(bool)})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Wait and verify PVC is fully deleted")
+		for _, volume := range volumesBeforeScaleDown {
+			ginkgo.By(fmt.Sprintf("Wait and verify PVC is fully deleted %s", volume))
+			err := fpv.DeletePersistentVolumeClaim(client, volume, namespace)
+			if !apierrors.IsNotFound(err) {
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+		}
+	})
+
+	ginkgo.It("Staging Should create and delete pod with the same volume source", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		var pvc *v1.PersistentVolumeClaim
+		var err error
+
+		ginkgo.By("Create PVC ")
+		storageclass, err := client.StorageV1().StorageClasses().Get(ctx, storagePolicyName, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		pvc, err = fpv.CreatePVC(client, namespace,
+			getPersistentVolumeClaimSpecWithStorageClass(namespace, diskSize, storageclass, nil, v1.ReadWriteOnce))
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By(fmt.Sprintf("Waiting for claim %s to be in bound phase", pvc.Name))
+		pvs, err := fpv.WaitForPVClaimBoundPhase(client, []*v1.PersistentVolumeClaim{pvc},
+			framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(pvs).NotTo(gomega.BeEmpty())
+		pv := pvs[0]
+		volumeID := pv.Spec.CSI.VolumeHandle
+
+		defer func() {
+			err := fpv.DeletePersistentVolumeClaim(client, pvc.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = e2eVSphere.waitForCNSVolumeToBeDeleted(pv.Spec.CSI.VolumeHandle)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Creating pod")
+		pod, err := createPod(client, namespace, nil, []*v1.PersistentVolumeClaim{pvc}, false, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
```

Comment:
Better to add defer after this Or create after each and handle this delete POD , if in case of any failure before the POD is getting deleted in line 857

Reply:
Done